src/docs/arch-fast-truncate.dox

```diff
@@ -0,0 +1,847 @@
+@row{btmem.h, Page header flag, \c WT_PAGE_FT_UPDATE,
+The on-disk flag that indicates the presence of page delete information in deleted-address
+cells.}
+@row{btmem.h, Update flag, \c WT_UPDATE_RESTORED_FAST_TRUNCATE,
+A flag used in prepare handling to identify updates from instantiation.}
+@row{btmem.h, Structure member, \c WT_PAGE_MODIFY::instantiated,
+The flag marking an in-memory \c WT_REF as an instantiated page.}
+@row{btmem.h, Structure member, \c WT_PAGE_MODIFY::inst_updates,
+The updates used to instantiate an unresolved truncation.}
+@row{btmem.h, Type, \c WT_PAGE_DELETED,
+The structure that holds page deletion information.}
+@row{btmem.h, Structure member, \c WT_REF::page_del,
+The page deletion information for a \c WT_REF.}
+
+@row{cell.h, Enumerator, \c WT_CELL_ADDR_DEL,
+The deleted-address cell type (along with the other cell types).}
+@row{cell.h, Misc, \c WT_CELL,
+The allocation of space in \c WT_CELL for the on-disk deletion information.}
+@row{cell.h, Structure member, \c WT_CELL_UNPACK_ADDR::page_del,
+The deletion information unpacked from an address cell.}
+
+@row{reconcile.h, Structure member, \c WT_CHILD_MODIFY_STATE::del,
+A \c WT_PAGE_DELETED that allows \c __wt_rec_child_modify to return page deleted
+information about a child ref after unlocking it.}
+
+@row{txn.h, Enumerator, \c WT_TXN_OP_REF_DELETE,
+The operation type for a fast-truncate transaction operation.}
+@row{txn.h, Structure member, \c WT_TXN_OP::ref,
+The data for a fast-truncate transaction operation: the \c WT_REF.}
+
+@row{btree_inline.h, Structure member, \c WT_ADDR_COPY::page_deleted,
+Page deletion information unpacked from the on-disk cell by \c __wt_ref_addr_copy.}
+@row{btree_inline.h, Structure member, \c WT_ADDR_COPY::del_set,
+True if page deletion information was unpacked.}
+@row{btree_inline.h, Hook, in \c __wt_ref_addr_copy,
+Return the page delete information unpacked from the address.}
+@row{btree_inline.h, Function, \c __wt_page_del_visible,
+Function for checking thread visibility of a \c WT_PAGE_DELETED.}
+@row{btree_inline.h, Function, \c __wt_page_del_visible_all,
+Function for checking global visibility of a \c WT_PAGE_DELETED.}
+@row{btree_inline.h, Function, \c __wt_page_del_committed,
+Function for checking whether a \c WT_PAGE_DELETED is committed.}
+@row{btree_inline.h, Function, \c __wt_btcur_skip_page,
+The page-skip function used by cursor next and previous.}
+
+@row{reconcile_inline.h, Hook, in \c __wt_rec_cell_build_addr,
+A hook to choose \c WT_CELL_ADDR_DEL cells when needed and propagate any passed-in page
+deletion information to the packing code.}
+
+@row{timestamp_inline.h, Macro, \c WT_TIME_AGGREGATE_UPDATE_PAGE_DEL,
+Akin to \c WT_TIME_AGGREGATE_UPDATE but for a \c WT_PAGE_DELETED; used in internal page
+reconciliation.}
+
+@row{txn_inline.h, Function, \c __wt_txn_op_delete_apply_prepare_state,
+The code for updating \c WT_PAGE_DELETED at prepare time.  updates to \c WT_PAGE_DELETED.}
```

Comment:
Is this correct?

Reply:
Apparently not :-) The second half appears to be some sort of leftover. Removed in [af26591](https://github.com/wiredtiger/wiredtiger/pull/8237/commits/af265919038ffe7086fbdc1db36c6800c63bc466)